libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1133,32 +1133,20 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 				") in value for variable assignment (0) needed"
 			);
 	}
-	else if (valueTypes.size() != variables.size())
+	else if (valueTypes.size() != variables.size() && !valueTypes.empty())
 	{
-		if (v050)
-			m_errorReporter.fatalTypeError(
-				_statement.location(),
-				"Different number of components on the left hand side (" +
-				toString(variables.size()) +
-				") than on the right hand side (" +
-				toString(valueTypes.size()) +
-				")."
-			);
-		else if (!variables.front() && !variables.back())
-			m_errorReporter.fatalTypeError(
-				_statement.location(),
-				"Wildcard both at beginning and end of variable declaration list is only allowed "
-				"if the number of components is equal."
-			);
-		else
-			m_errorReporter.warning(
-				_statement.location(),
-				"Different number of components on the left hand side (" +
-				toString(variables.size()) +
-				") than on the right hand side (" +
-				toString(valueTypes.size()) +
-				")."
-			);
+		// XXX If the RHS is "void", then we already have an error generated elsewhere,
```

Comment:
Please don't use `XXX`. Can you perhaps replace this large comment by an assertion that checks that if the RHS is void, the error reporter has at least one error?